SonarQube.Common/ConsoleColorScope.cs

```diff
@@ -0,0 +1,80 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ConsoleColorScope.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Diagnostics;
+
+namespace SonarQube.Common
+{
+    /// <summary>
+    /// Utility class that changes the console text colour for the lifetime of the instance
+    /// </summary>
+    /// <remarks>This will have no effect if the console output streams have been re-directed</remarks>
+    internal class ConsoleColorScope : IDisposable
+    {
+        private readonly ConsoleColor originalForeground;
+        private readonly ConsoleColor originalBackground;
+
+        public ConsoleColorScope(ConsoleColor textColor)
+        {
+            this.originalForeground = Console.ForegroundColor;
+            this.originalBackground = Console.BackgroundColor;
+
+            // Check the text doesn't clash with the background color
+            ConsoleColor newBackground = Console.BackgroundColor;
```

Comment:
Is this necessary / did you encounter this situation in practice ? Yellow text on white background is also "clashing" but it would be difficult to correct all colour schemes.